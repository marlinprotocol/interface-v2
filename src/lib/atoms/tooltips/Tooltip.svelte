<script lang="ts">
	import { staticImages } from '$lib/components/images/staticImages';
	import type { TooltipPlacement } from '$lib/types/componentTypes';
	import { cn } from '$lib/utils/helpers/commonHelper';

	export let placement: TooltipPlacement = 'top';
	export let styleClass: string = '';

	const tooltipPosition = {
		top: {
			container: '-top-5 left-1/2 -translate-x-1/2 -translate-y-full',
			arrow:
				'after:left-1/2 after:-translate-x-1/2 after:-translate-y-1/2 top-full left-1/2 -ml-[15px] w-[30px] h-[15px]'
		},
		bottom: {
			container: 'top-10 left-1/2 -translate-x-1/2 translate-y-0',
			arrow:
				'after:left-1/2 after:-translate-x-1/2 after:translate-y-1/2 bottom-full left-1/2 -ml-[15px] w-[30px] h-[15px]'
		},
		left: {
			container: 'top-1/2 right-full mr-5 translate-x-0 -translate-y-1/2',
			arrow:
				'after:left-0 after:top-1/2 after:-translate-x-1/2 after:-translate-y-1/2 left-full top-1/2 -mt-[15px] w-[15px] h-[30px]'
		},
		right: {
			container: 'top-1/2 left-full ml-5 translate-x-0 -translate-y-1/2',
			arrow:
				'after:left-0 after:top-1/2 after:translate-x-1/2 after:-translate-y-1/2 right-full top-1/2 -mt-[15px] w-[15px] h-[30px]'
		}
	};
</script>

<div
	class={cn('group relative inline-block cursor-pointer text-left', styleClass)}
	data-testid="tooltip"
>
	<!-- adds a default icon for tooltip when no icon is passed to the slot -->
	{#if !$$slots.tooltipIcon}
		<img src={staticImages.alertV2Icon} alt="Info" width="16px" class="icon-invert" />
	{/if}
	<slot name="tooltipIcon" />
	<div
		class={cn(
			'absolute z-100 box-border hidden w-max max-w-[320px] rounded-xl bg-secondary-content p-3 shadow-[0px_12px_16px_-4px_#10182814] group-hover:block',
			tooltipPosition[placement].container
		)}
	>
		<slot name="tooltipContent" />
		<i
			class={cn(
				'absolute overflow-hidden after:absolute after:h-[15px] after:w-[15px] after:rotate-45 after:bg-secondary-content after:shadow-[0px_12px_16px_-4px_#10182814]',
				tooltipPosition[placement].arrow
			)}
		></i>
		<div class="after:left-0 after:top-1/2 after:-translate-y-1/2 after:translate-x-1/2"></div>
	</div>
</div>
