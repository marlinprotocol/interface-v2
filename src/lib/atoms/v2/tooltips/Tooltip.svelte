<script lang="ts">
	import type { TooltipDirection, TooltipVariant } from '$lib/types/componentTypes';

	export let tooltipText: string;
	export let styleClass = '';
	export let tooltipVariant: TooltipVariant = 'tooltip-secondary';
	export let tooltipDirection: TooltipDirection = 'tooltip-top';

	$: tooltipClass = `${styleClass} ${tooltipVariant} ${tooltipDirection}`;
</script>

<div
	data-testid="tooltip"
	class="tooltip inline-flex text-left font-normal {tooltipClass}"
	data-tip={tooltipText}
>
	<slot />
</div>
