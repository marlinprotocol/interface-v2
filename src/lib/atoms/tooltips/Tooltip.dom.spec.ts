import { cleanup, render } from '@testing-library/svelte';
import Tooltip from './Tooltip.svelte';
import html from '@playpilot/svelte-htm';

describe('Tooltip', () => {
	afterEach(() => {
		cleanup();
		vi.resetAllMocks();
	});

	it('renders with default attribute', async () => {
		const { findByText, findByTestId, component } = render(html`
                <${Tooltip}>
                 <div slot='tooltipContent' class="tooltip-content">This is tooltip content</div>
                </${Tooltip}>
              `);
		expect(await findByText('This is tooltip content')).toBeTruthy();
		expect(await findByTestId('tooltip')).toBeTruthy();
		expect(
			(await findByTestId('tooltip')).getElementsByClassName('-top-5 left-1/2')[0]
		).toBeTruthy();
		expect((await findByTestId('tooltip')).getElementsByClassName('top-full')[0]).toBeTruthy();
		expect(component).toMatchSnapshot();
	});
});
