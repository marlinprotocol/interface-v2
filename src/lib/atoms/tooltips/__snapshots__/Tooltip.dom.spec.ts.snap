// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tooltip > renders with default attribute 1`] = `
Component {
  "$$": {
    "after_update": [],
    "before_update": [],
    "bound": {},
    "callbacks": {},
    "context": Map {},
    "ctx": [
      [Function],
      {},
      [
        [Function],
      ],
      {},
    ],
    "dirty": [
      -1,
    ],
    "fragment": {
      "c": [Function],
      "d": [Function],
      "m": [Function],
      "p": [Function],
    },
    "not_equal": [Function],
    "on_destroy": [
      [Function],
    ],
    "on_disconnect": [],
    "on_mount": [],
    "props": {},
    "root": <div>
      <div
        class="group relative inline-block cursor-pointer text-left"
        data-testid="tooltip"
      >
        <img
          alt="Info"
          class="icon-invert"
          src="/images/alertV2Icon.svg"
          width="16px"
        />
         
         
        <div
          class="absolute z-100 box-border hidden w-max max-w-[320px] rounded-xl bg-secondary-content p-3 shadow-[0px_12px_16px_-4px_#10182814] group-hover:block -top-5 left-1/2 -translate-x-1/2 -translate-y-full"
        >
          <div
            class="tooltip-content"
          >
            This is tooltip content
          </div>
           
          <i
            class="absolute overflow-hidden after:absolute after:h-[15px] after:w-[15px] after:rotate-45 after:bg-secondary-content after:shadow-[0px_12px_16px_-4px_#10182814] after:left-1/2 after:-translate-x-1/2 after:-translate-y-1/2 top-full left-1/2 -ml-[15px] w-[30px] h-[15px]"
          />
           
          <div
            class="after:left-0 after:top-1/2 after:-translate-y-1/2 after:translate-x-1/2"
          />
        </div>
      </div>
      <!--&lt;Tooltip&gt;-->
    </div>,
    "skip_bound": false,
    "update": [Function],
  },
  "$$set": undefined,
}
`;
